prospector/commit_preprocessor/test_preprocessor.py

```diff
@@ -41,8 +52,77 @@ def test_proprocess_commit_set(repository):
     assert len(preprocessed_commits) == len(commit_set)
 
 
+def test_extract_ghissue_references():
+    result = extract_ghissue_references(
+        "#1 blah #2 blah a#3 blah #4b blah c#5d blah 0#6 blah #10 44 #999 0 #6"
+    )
+    assert result == ["#1", "#2", "#3", "#4", "#5", "#6", "#10", "#999", "#6"]
+
+
+def test_extract_jira_references():
+    result = extract_jira_references(
+        "FIRST-1 blah second-2 blah thIrD-3 blah FOURTH-4b blah fIftH-5d blah TENTH-10 44 NiNeNinENinE-999 0 sIx-6"
+    )
+    assert result == [
```

Comment:
the project ID is always capital letters: sIx-6 should not be matched for example

Reply:
The current implementation of extract_jira_references accepts all these project IDs. So its implementation needs to be adjusted with the test as well. Could you confirm what should be the exact result list for this test input? Isn't thIrD-3 and other non capital letter IDs should be rejected as well?

Reply:
Yes, see this as an example: https://github.com/apache/camel/commit/1df559649a96a1ca0368373387e542f46e4820da

At least, I have never seen lowercase characters in those JIRA IDs